tests/test_internals/test_si.py

```diff
@@ -293,7 +293,8 @@ def test_write_sorting(self):
 
             # check that missing props are filled correctly
             np.testing.assert_array_equal(nwbfile.units["float_prop"][:], [80.0, np.nan, np.nan])
-            np.testing.assert_array_equal(nwbfile.units["int_prop"][:], [80.0, np.nan, np.nan])
+            # int is not filled with nan in SI
+            # np.testing.assert_array_equal(nwbfile.units["int_prop"][:], [80.0, np.nan, np.nan])
```

Comment:
```suggestion
            # np.testing.assert_array_equal(nwbfile.units["int_prop"][:], [80, 0, 0])
```
Should update test to reflect and communicate new standard default.

@bendichter Note that this means all integer valued properties have a default fill value of `0` now, is this OK with you? If not, an alternative for us in this PR would be to restore a part of the aggregated unit code, and coerce integer valued properties to float in order to designate items truly set as `0` from missing values filled by `np.nan`.

Reply:
@CodyCBakerPhD we could also add the option `missing_value` to Spikeinterface when setting property on a subset of channels. In case one uses `np.nan`, then the property becomes float, but this would leave to possibility to set say -1 as default missing option. What do you think?

Reply:
Talked to Ben about it and he supports the `float` casting of `int` types. Increases data sizes a tad, but this isn't that heavy a data type anyway. Really we would want to use something like numpy masks to distinguish missing values in all cases, but NWB doesn't really have a great analog of that, although it is important for us to properly distinguish missing vs. non-missing data values.

@alejoe91 At what level do you think we should resolve that? In this PR, or in SI?

Reply:
I think in SI, will try to implement the `missing_value` option tomorrow :) Will ping you when I'm done, thanks for the reminder

Reply:
@CodyCBakerPhD to be tested with this patch: https://github.com/SpikeInterface/spikeinterface/pull/483

Now we can provide the `missing_value` argument when setting property. Note that setting `np.nan` for an int property will result in the property values being handled as `object` array, which gives more flexibility (e.g. you can have a nan value in a string array, etc.)

Reply:
@CodyCBakerPhD after discussion with Sam, we decided to be more strict about missing values. If you set an int property, the `missing_value` must also be int. In case one has int values and wants to set nans as missing, she should cast the property values outside of the property function. I think that this stricter approach will be less error-prone and "magic" in the long run

Reply:
I disagree. There is no int value that works for missing values.

Reply:
@alejoe91 @samuelgarcia I do agree with Ben as per my justifications above, but I see what you're saying about giving the user control via requiring them to set it as float if they want to do that themselves. The thing is, they have to think about that pretty far in advance, whereas it would be the kind of thing that is easy to slip the mind unless you remember every time you add a property to (a) check if you have missing values and (b) remember that, in SI, the way you specify missing values is to pre-cast as float. I'd recommend just changing line #163 in the BaseExtractor (https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/base.py#L163) to `if values.dtype.kind in ['f', 'i']:` if I were you guys.

If you don't budge on that though, we should still have code on this PR that handles the float upcasting.

Reply:
Hi @CodyCBakerPhD @alejoe91 @bendichter.
I think that in the spikeinterface context we should not change the dtype on the fly and magically depending missing values. When an int is set to property, generally it is on purpose (indexing).
Having missing values, in context of spike sorting is a corner case contary to data management.
For property handling we have a big constrain of dump to dict  for parralel processing and dump to json for provenance. Handling in missing memory is easy, we could use mask array of numpy but the dump to json will be pain.
This is why I suggest that in SI, there is no magic upcasting on the fly. An internal property should keep the same dtype.
Upcasting should be external.
So yes it will force us to have some code here in conversion tools but I think it is a good practice to have clear rules for upcasting here and not in SI.

Reply:
But for now, I'll just get something in this PR to fix the issue temporarily.

Reply:
Furthermore, from lines https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/base.py#L157-L170 I don't actually see a way for us to take a Sorting object this far downstream and handle this issue on our side, since by that time the information of both the `ids` set at function call and original `values` aren't kept anywhere and so the information of missing values is lost.

This is a step back even from the old `SortingExtractors`, where these were dynamic dict-of-dicts that allowed not only allowed users full control over missing values, but also preserved that information internally so it could be used downstream.

@alejoe91 Just so we can get this PR done, what I would recommend in the meantime is for you to restore some of the previous code for inferring missing data for integer values properties in the case where the object passed was a `SortingExtractor`, similar to (and likely containable within) the block of code at the end that handles the `unit_spike_features`; just so we can still maintain support for that via old `SortingExtractor` objects until the new SI makes the masking changes.